cmd/bbolt/main_test.go

```diff
@@ -135,36 +135,70 @@ func TestPageCommand_Run(t *testing.T) {
-	err := db.Fill([]byte("data"), 1, 100,
-		func(tx int, k int) []byte { return []byte(fmt.Sprintf("key_%d", k)) },
-		func(tx int, k int) []byte { return []byte(fmt.Sprintf("value_%d", k)) },
-	)
-	require.NoError(t, err)
+	t.Run("test data in string format", func(t *testing.T) {
+		db := btesting.MustCreateDBWithOption(t, &bolt.Options{PageSize: 4096})
+		srcPath := db.Path()
+
+		// Insert some sample data
+		t.Log("Insert some sample data")
+		err := db.Fill([]byte("data"), 1, 100,
+			func(tx int, k int) []byte { return []byte(fmt.Sprintf("key_%d", k)) },
+			func(tx int, k int) []byte { return []byte(fmt.Sprintf("value_%d", k)) },
+		)
+		require.NoError(t, err)
 
-	defer requireDBNoChange(t, dbData(t, srcPath), srcPath)
+		defer requireDBNoChange(t, dbData(t, srcPath), srcPath)
+
+		meta := readMetaPage(t, srcPath)
+		leafPageId := 0
+		for i := 2; i < int(meta.Pgid()); i++ {
+			p, _, err := guts_cli.ReadPage(srcPath, uint64(i))
+			require.NoError(t, err)
+			if p.IsLeafPage() && p.Count() > 1 {
+				leafPageId = int(p.Id())
+			}
+		}
+		require.NotEqual(t, 0, leafPageId)
 
-	meta := readMetaPage(t, srcPath)
-	leafPageId := 0
-	for i := 2; i < int(meta.Pgid()); i++ {
-		p, _, err := guts_cli.ReadPage(srcPath, uint64(i))
+		m := NewMain()
+		err = m.Run("page-item", db.Path(), fmt.Sprintf("%d", leafPageId), "0")
 		require.NoError(t, err)
-		if p.IsLeafPage() && p.Count() > 1 {
-			leafPageId = int(p.Id())
+		if !strings.Contains(m.Stdout.String(), "key_0") || !strings.Contains(m.Stdout.String(), "value_0") {
+			t.Fatalf("Unexpected output:\n%s\n", m.Stdout.String())
 		}
-	}
-	require.NotEqual(t, 0, leafPageId)
+	})
 
-	m := NewMain()
-	err = m.Run("page-item", db.Path(), fmt.Sprintf("%d", leafPageId), "0")
-	require.NoError(t, err)
-	if !strings.Contains(m.Stdout.String(), "key_0") || !strings.Contains(m.Stdout.String(), "value_0") {
-		t.Fatalf("Unexpected output:\n%s\n", m.Stdout.String())
-	}
+	t.Run("test data in int64 format", func(t *testing.T) {
+		db := btesting.MustCreateDBWithOption(t, &bolt.Options{PageSize: 4096})
+		srcPath := db.Path()
+
+		// Insert some sample data
+		t.Log("Insert some sample data")
+		err := db.Fill([]byte("data"), 1, 100,
+			func(tx int, k int) []byte { return convertInt64IntoBytes(t, 10001) },
+			func(tx int, k int) []byte { return convertInt64IntoBytes(t, 10002) },
+		)
+		require.NoError(t, err)
+
+		defer requireDBNoChange(t, dbData(t, srcPath), srcPath)
+
+		meta := readMetaPage(t, srcPath)
+		leafPageId := 0
+		for i := 2; i < int(meta.Pgid()); i++ {
+			p, _, err := guts_cli.ReadPage(srcPath, uint64(i))
+			require.NoError(t, err)
+			if p.IsLeafPage() && p.Count() > 1 {
+				leafPageId = int(p.Id())
+			}
+		}
+		require.NotEqual(t, 0, leafPageId)
+		m := NewMain()
+		err = m.Run("page-item", db.Path(), fmt.Sprintf("%d", leafPageId), "0")
+		require.NoError(t, err)
+		if !strings.Contains(m.Stdout.String(), "10001") || !strings.Contains(m.Stdout.String(), "10002") {
```

Comment:
This isn't correct. When the key or value isn't printable, the output will be hex format string.